### DISCUSSION TITLE
Correcting the original translation

### DISCUSSION THREAD

**Discussion by Laserlicht** [ihhub/fheroes2#6444] [Ideas]
In the German translation, umlauts are represented with substitute characters within the maps and campaigns. This is of course not so nice.

Is there any way we can correct this? Unfortunately, this does not work fully automatically. Manual interventions are always necessary.

Example: `zurueckzuobern `-> `zurückzuerobern` (first `ue` -> `ü`, second not).

Would need to be adjusted: `ue` -> `ü`, `ae` -> `ä`, `oe` -> `ö`, `Ue` -> `Ü`, `Ae` -> `Ä`, `Oe` -> `Ö`, `ss` -> `ß`

Adjusting the files would be possible (umlauts seem to work in maps). But then you can not deliver because of the copyright. Patches I would not find good.

I don't know if this is also a problem in other languages. Could a replacement table be introduced for such cases?

See attachment. There are all (incorrect) German words (from the maps and campaigns) in it, and how they should be replaced.

[wordlist_de.csv](https://github.com/ihhub/fheroes2/files/10329761/wordlist_de.csv)

**Comment by undef21** [comment]
Hello @Laserlicht. The original german localization has additional symbols from the german alphabet, but AFAIK these symbols aren't used in the game and original maps. I found only a few fan maps which have umlauts and  eszett. You can check and replace correct chars in the game interface, but map messages required  another approach, eg. creating a translate file for each map (something likes `Alteris.mp2.de`).

**Comment by zenseii** [comment]
Hi, @Laserlicht and @undef21. This should be possible to do, but it would require some new logic that for when German is selected all those digraphs (ue, oe etc.) will be replaced.

Are there no instances in which such two letters should not be replaced in the German language/German maps?

**Reply by Laserlicht** [reply]
@zenseii Yes, as I said. It cannot be done automatically. Since there are some two letter combinations that need to be replaced and others that do not.

@undef21 Yes, truthfully the approach with `Alteris.mp2.de` would be best. The question is what would the file look like? A simple replacement list?

**Reply by undef21** [reply]
Hello @Laserlicht.

>The question is what would the file look like? A simple replacement list?

IMHO, yes.

Actually there are at least three types of text. The first type is messages from `de.po`. These messages could be fixed right now by reviewing `files/lang/de.po`. The second type is messages on game sprites. These messages are parts of images and could be changed at runtime. The third type is messages inside maps. The approach with `Alteris.mp2.de` is a possible solution for this case.
